pkg/provider/helm/values_test.go

```diff
@@ -982,6 +988,7 @@ func TestMergeMaps(t *testing.T) {
 									"enabled": true,
 								},
 							},
+							"clusterStaticEntries": Values{},
```

Comment:
Do we never expect a value in any of these test cases? What about then the spire agent gets flipped to the deterministic SPIFFE ID case per the logic above?

Reply:
Just added a test case for this, thanks 👌